jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/constraint/ServiceLoadActivityLevelConstraint.java

```diff
@@ -71,12 +68,22 @@ public ConstraintsStatus fulfilled(JobInsertionContext iFacts, TourActivity prev
             }
 
         }
-        if (newAct instanceof PickupServiceDEPRECATED || newAct instanceof ServiceActivityNEW) {
+
+        if (newAct.getSize().isPositive()) {
             if (!Capacity.addup(newAct.getSize(), futureMaxLoad).isLessOrEqual(iFacts.getNewVehicle().getType().getCapacityDimensions())) {
                 return ConstraintsStatus.NOT_FULFILLED;
             }
         }
-        if (newAct instanceof DeliverServiceDEPRECATED) {
+
+        /*
+         * REMARK - Balage - This negating could be a bottleneck if called too
+         * many times. Has to be mesured. If rational, the activities could
+         * store their size as absolute value as well. (We could rename
+         * getSize() as getCargoChange(), and the absolute value as
+         * getCargoSize(). For positive or zero activities as Service and Pickup
+         * they could refer to the same object.)
+         */
```

Comment:
It's ready (I only have to do testing.)

I did synthetic benchmarking for 16 million calls.
**Add:**
Old: 20.2 sec
New: 6.32 sec
Increase: 3.6x

**Subtract:**
Old: 20.0 sec
New: 6.42 sec
Increase: 3.4x

Strangely in real life test (PickAndDeliveryExample) there is no performance increase, although it calls 57 million times these functions. I tried to keep it as static methods, I tried to move it into normal method. Because I can't give a reason for this strange behavoiur, I guess the solution is somewhere in my mesuring method.